src/jit/rangecheck.cpp

```diff
@@ -1034,17 +1044,22 @@ bool RangeCheck::DoesVarDefOverflow(BasicBlock* block, GenTreePtr stmt, GenTreeP
     }
     // Get the parent node which is an asg.
     GenTreePtr asg = loc->parent;
-    assert(asg->OperKind() & GTK_ASGOP);
+    assert(asg->OperIsAssignment());
     switch (asg->OperGet())
     {
         case GT_ASG:
             return DoesOverflow(loc->block, loc->stmt, asg->gtGetOp2(), path);
 
+#ifdef LEGACY_BACKEND
         case GT_ASG_ADD:
             // For GT_ASG_ADD, op2 is use, op1 is also use since we side table for defs in useasg case.
             return DoesBinOpOverflow(loc->block, loc->stmt, asg->gtGetOp1(), asg->gtGetOp2(), path);
+#endif
 
         default:
+#ifndef LEGACY_BACKEND
+            unreached();
```

Comment:
Note that this causes a desktop build break due to unreachable code warning

Reply:
Hmm, what's different between the desktop build and the CoreCLR build? Compilation options?

Reply:
Possibly. And different toolsets.

Reply:
It looks like one difference is that the desktop build uses /W4 while the CoreCLR build uses /W3